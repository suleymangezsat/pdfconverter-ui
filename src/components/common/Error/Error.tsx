import { Box, Button, Typography } from "@material-ui/core";
import { memo, ReactElement } from "react";

type Props = {
  onRefresh: () => void;
};

export const Error = memo(
  ({ onRefresh }: Props): ReactElement => (
    <Box textAlign="center">
      <Typography>Something went wrong</Typography>
      <Box m={1}>
        <Button variant="outlined" onClick={() => onRefresh()}>
          Try Again
        </Button>
      </Box>
    </Box>
  )
);
