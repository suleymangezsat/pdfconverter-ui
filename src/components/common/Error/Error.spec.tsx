import React from "react";
import { Error } from "./Error";
import { shallow } from "enzyme";

describe("components/common/Error", () => {
  const handleRefresh = jest.fn();

  it("should render Error component correctly", () => {
    const component = shallow(<Error onRefresh={handleRefresh} />);
    expect(component).toMatchSnapshot();
  });
  it("should trigger onRefresh", () => {
    const component = shallow(<Error onRefresh={handleRefresh} />);
    component.find("WithStyles(ForwardRef(Button))").simulate("click");
    expect(handleRefresh).toHaveBeenCalled();
  });
});
