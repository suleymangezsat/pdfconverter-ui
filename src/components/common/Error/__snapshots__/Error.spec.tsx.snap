// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Error should render Error component correctly 1`] = `
<Styled(MuiBox)
  textAlign="center"
>
  <WithStyles(ForwardRef(Typography))>
    Something went wrong
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox)
    m={1}
  >
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="outlined"
    >
      Try Again
    </WithStyles(ForwardRef(Button))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
